api/v1alpha1/externaldns_webhook.go

```diff
@@ -129,3 +149,22 @@ func (r *ExternalDNS) validateProviderCredentials() error {
 	}
 	return nil
 }
+
+// Returns true if platform is OCP
+func isOCP(kubeClient discovery.DiscoveryInterface) bool {
+	// Since, CRD for OpenShift API Server was introduced in OCP v4.x we can verify if the current cluster is on OCP v4.x by
+	// ensuring that resource exists against Group(operator.openshift.io), Version(v1) and Kind(OpenShiftAPIServer)
+	// In case it doesn't exist we assume that external dns is running on non OCP 4.x environment
+	resources, err := kubeClient.ServerResourcesForGroupVersion(group + "/" + version)
+	fmt.Printf("\nresources %v", resources)
```

Comment:
The logging should be done with the logger from the controller runtime.

Reply:
actually logging it is not needed. removed it.